addons/account/models/account_move.py

```diff
@@ -1942,6 +1943,28 @@ def _compute_tax_country_code(self):
         for record in self:
             record.tax_country_code = record.tax_country_id.code
 
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    def _compute_duplicated_ref(self):
+        """ The computed field only appears in the move view form and thus is supposed
+        to only be computed for one move at a time which makes the search in loop "ok"
+        """
+        for move in self:
+            if not (move.is_purchase_document() and move.ref):
+                move.duplicated_ref = self.env['account.move']
+                continue
+
+            duplicated_ref_domain = [
+                ('company_id', '=', move.company_id.id),
+                ('move_type', '=', move.move_type),
+                ('commercial_partner_id', '=', move.commercial_partner_id.id),
+                ('ref', '=', move.ref),
+                ('invoice_date', '=', move.invoice_date or False),
```

Comment:
If this is non blocking, we should remove the filtering on the date IMO

Reply:
I disagree with that. What makes an invoice unique is the combination of (partner, date, ref).
An accountant which doesn't go by the "put the bill ref as label" standard (and I worked for some of them) may just put "rent"/"machine"/whatever in the ref of the move just to know what the billl is related to.

Removing the date will then make this bill appears as a duplicate while it is not because it's another bill. (and you know that because the date isn't the same)

Reply:
> An accountant which doesn't go by the "put the bill ref as label" standard (and I worked for some of them) may just put "rent"/"machine"/whatever in the ref of the move just to know what the billl is related to.

But then if they buy two similar things on the same day, they will have the same issue...
They are not using the software like intended, and if they want to use it like that, they could just append the date:
"rent January 2022"